src/Entities/Player.cpp

```diff
@@ -2086,6 +2083,9 @@ bool cPlayer::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 		// New world will take over and announce client at its next tick
 		auto PlayerPtr = static_cast<cPlayer *>(ParentChunk->RemoveEntity(*this).release());
 		a_World->AddPlayer(std::unique_ptr<cPlayer>(PlayerPtr), &a_OldWorld);
+
+		// Broadcast the player into the new world.
+		a_World->BroadcastSpawnEntity(*this);
```

Comment:
(Just a quick question) 
Shouldn't it be actually moved to where `a_World->AddPlayer()` does the actual work of adding the player? Possibly in the `a_World`'s own thread, instead of the old world's tick thread.

Reply:
Seems like a much cleaner solution, and would ensure that entity spawns are always broadcasted.